src/kernel/pm/semaphore.c

```
@@ -239,6 +239,40 @@ int semaphore_delete(int semid)
     return (-1);
 }
 
+/*
+ * @brief Return value of semaphore count.
+ *
+ * @p semid Semaphore id.
+ *
+ * @return (Semaphore count) if successful , (-1) otherwise.
+ */
+int semaphore_getcount(int semid)
+{
+    // Verify if semaphore is SEMAPHORE_ACTIVE.
+    if (!is_semaphore_SEMAPHORE_ACTIVE(semid)) {
+        return (-1);
+    }
+
+    // Verify if semaphore is get.
+    if (!is_semaphore_get(semid)) {
+        return (-1);
```

Could we return a negative error code instead?

---

Yes, i agree.